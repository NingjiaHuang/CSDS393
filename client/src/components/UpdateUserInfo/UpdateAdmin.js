import React, { Fragment, useState } from "react";
import { toast } from "react-toastify";
import 'react-toastify/dist/ReactToastify.css'

const UpdatePP = (admini) => {
    const [inputs, setInputs] = useState({
        username: admini.username,
        user_password: admini.user_password,
        reg_email: admini.reg_email,
        reg_phone: admini.reg_phone
    }); 

    const {username, user_password, reg_email, reg_phone} = inputs;
    const account_type = "admini";
    
    const onChange = (e) => {
        setInputs({...inputs, [e.target.name]:e.target.value})
    }

    const onSubmitForm = async(e) => {
        e.preventDefault();
        try{
            const body = {username, user_password, account_type, reg_email, reg_phone};
            
            const response = await fetch("http://localhost:4020/auth/register/admini", //update potential parents
            {
                method: "PUT",
                headers: {"Content-Type": "application/json"},
                body: JSON.stringify(body)
            });
        }catch(err){
            console.error(err.message)
        }
    }

    return (
        <Fragment>
            <h1 className="text-center my-5">Update your account</h1>
            <form onSubmit={onSubmitForm}>
                <label>Username: </label><input type="text" name="username"  value={username} placeholder={admini.username} onChange={e => onChange(e)} className="form-control my-2"/>
                <label>Password: </label> <input type="text" name="password" placeholder="password" className="form-control my-2" value={user_password}  placeholder={admini.user_password} onChange={e => onChange(e)}/>
                <label>Email: </label><input type="text" name="reg_email" placeholder="email" className="form-control my-3" value={reg_email} placeholder={admini.reg_email} onChange={e => onChange(e)}/>
                <label>Phone: </label><input type="text" name="reg_phone" placeholder="phone" className="form-control my-3" value={reg_email} placeholder={admini.reg_email} onChange={e => onChange(e)}/>                <button className="btn btn-success btn-block">Update</button>
            </form>
        </Fragment>
    )
}

export default UpdatePP;

