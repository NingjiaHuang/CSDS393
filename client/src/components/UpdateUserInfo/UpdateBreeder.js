import React, { Fragment, useState } from "react";
import { toast } from "react-toastify";
import 'react-toastify/dist/ReactToastify.css'

const UpdateBreeder = (breeder) => {
    const [inputs, setInputs] = useState({
        username: breeder.username,
        user_password: breeder.user_password,
        reg_email: breeder.reg_email,
        reg_phone: breeder.reg_phone,
        cattery_name:breeder.cattery_name,
        organization: breeder.organization,
        owner_name:breeder.owner_name,
        city: breeder.city
    }); 

    const {username, user_password, reg_email, reg_phone, cattery_name,organization, owner_name, city } = inputs;
    const account_type = "breeder";
    
    const onChange = (e) => {
        setInputs({...inputs, [e.target.name]:e.target.value})
    }

    const onSubmitForm = async(e) => {
        e.preventDefault();
        try{
            const body =  {username, user_password, reg_email, reg_phone, cattery_name,organization, owner_name, city };
            
            const response = await fetch("http://localhost:4020/auth/register/breeder", //update potential parents
            {
                method: "PUT",
                headers: {"Content-Type": "application/json"},
                body: JSON.stringify(body)
            });
        }catch(err){
            console.error(err.message)
        }
    }

    return (
        <Fragment>
            <h1 className="text-center my-5">Update your account</h1>
            <form onSubmit={onSubmitForm}>
                <label>Username: </label><input type="text" name="username"  value={username} placeholder={breeder.username} onChange={e => onChange(e)} className="form-control my-2"/>
                <label>Password: </label> <input type="text" name="password"className="form-control my-2" value={user_password}  placeholder={breeder.user_password} onChange={e => onChange(e)}/>
                <label>Email: </label><input type="text" name="reg_email"  className="form-control my-3" value={reg_email} placeholder={breeder.reg_email} onChange={e => onChange(e)}/>
                <label>Phone: </label><input type="text" name="reg_phone" className="form-control my-3" value={reg_email} placeholder={breeder.reg_email} onChange={e => onChange(e)}/>
                <label>Cattery name: </label><input type="text" name="cattery_name" className="form-control my-3" value={cattery_name} placeholder={breeder.cattery_name} onChange={e => onChange(e)}/>
                <label>Organization: </label><input type="text" name="organization" className="form-control my-3" value={organization} placeholder={breeder.organization} onChange={e => onChange(e)}/>
                <label>Owner name: </label><input type="text" name="owner_name" className="form-control my-3" value={owner_name} placeholder={breeder.owner_name} onChange={e => onChange(e)}/>
                <label>City: </label><input type="text" name="city" className="form-control my-3" value={city} placeholder={breeder.city} onChange={e => onChange(e)}/>
                <button className="btn btn-success btn-block">Update</button>
            </form>
        </Fragment>
    )
}

export default UpdateBreeder;

