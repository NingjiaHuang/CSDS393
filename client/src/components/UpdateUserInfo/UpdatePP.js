import React, { Fragment, useState } from "react";
import { toast } from "react-toastify";
import 'react-toastify/dist/ReactToastify.css'

const UpdatePP = (potential_parent) => {
    const [inputs, setInputs] = useState({
        username: potential_parent.username,
        user_password: potential_parent.user_password,
        reg_email: potential_parent.reg_email,
        reg_phone: potential_parent.reg_phone,   
        preferred_name: potential_parent.preferred_name
    }); 

    const {username, user_password, reg_email, reg_phone, preferred_name} = inputs;
    const account_type = "potential_parent";
    
    const onChange = (e) => {
        setInputs({...inputs, [e.target.name]:e.target.value})
    }

    const onSubmitForm = async(e) => {
        e.preventDefault();
        try{
            const body = {username, user_password, account_type, reg_email, reg_phone, preferred_name};
            
            const response = await fetch("http://localhost:4020/auth/register/potential_parents", //update potential parents
            {
                method: "PUT",
                headers: {"Content-Type": "application/json"},
                body: JSON.stringify(body)
            });
        }catch(err){
            console.error(err.message)
        }
    }

    return (
        <Fragment>
            <h1 className="text-center my-5">Update your account</h1>
            <form onSubmit={onSubmitForm}>
                <label>Username: </label><input type="text" name="username"  value={username} placeholder={potential_parent.username} onChange={e => onChange(e)} className="form-control my-2"/>
                <label>Password: </label> <input type="text" name="password" placeholder="password" className="form-control my-2" value={user_password}  placeholder={potential_parent.user_password} onChange={e => onChange(e)}/>
                <label>Email: </label><input type="text" name="reg_email" placeholder="email" className="form-control my-3" value={reg_email} placeholder={potential_parent.reg_email} onChange={e => onChange(e)}/>
                <label>Phone: </label><input type="text" name="reg_phone" placeholder="phone" className="form-control my-3" value={reg_email} placeholder={potential_parent.reg_email} onChange={e => onChange(e)}/>
                <label>Preferred name: </label><input type="text" name="preferred_name" placeholder="preferred name" className="form-control my-3" value={preferred_name} placeholder={potential_parent.preferred_name} onChange={e => onChange(e)}/>
                <button className="btn btn-success btn-block">Update</button>
            </form>
        </Fragment>
    )
}

export default UpdatePP;

