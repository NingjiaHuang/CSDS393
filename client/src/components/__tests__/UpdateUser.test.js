import {render, screen, cleanup} from '@testing-library/react';
import UpdateAdmin from '../UpdateUserInfo/UpdateAdmin';
import UpdateBreeder from '../UpdateUserInfo/UpdateBreeder';
import UpdatePP from '../UpdateUserInfo/UpdatePP';


import renderer from 'react-test-renderer'
import '@testing-library/jest-dom/extend-expect';

afterEach(() =>{
    cleanup();
})


test('matches snapshot of update admin info', () =>{
    const tree = renderer.create(<UpdateAdmin/>).toJSON();
    expect(tree).toMatchSnapshot();
})


test('matches snapshot of update breeder info', () =>{
    const tree = renderer.create(<UpdateBreeder/>).toJSON();
    expect(tree).toMatchSnapshot();
})

test('matches snapshot of update parent info', () =>{
    const tree = renderer.create(<UpdatePP/>).toJSON();
    expect(tree).toMatchSnapshot();
})

