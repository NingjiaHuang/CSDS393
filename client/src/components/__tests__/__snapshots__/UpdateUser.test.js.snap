// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot of update admin info 1`] = `
Array [
  <h1
    className="text-center my-5"
  >
    Update your account
  </h1>,
  <form
    onSubmit={[Function]}
  >
    <label>
      Username: 
    </label>
    <input
      className="form-control my-2"
      name="username"
      onChange={[Function]}
      type="text"
    />
    <label>
      Password: 
    </label>
     
    <input
      className="form-control my-2"
      name="password"
      onChange={[Function]}
      type="text"
    />
    <label>
      Email: 
    </label>
    <input
      className="form-control my-3"
      name="reg_email"
      onChange={[Function]}
      type="text"
    />
    <label>
      Phone: 
    </label>
    <input
      className="form-control my-3"
      name="reg_phone"
      onChange={[Function]}
      type="text"
    />
                    
    <button
      className="btn btn-success btn-block"
    >
      Update
    </button>
  </form>,
]
`;

exports[`matches snapshot of update breeder info 1`] = `
Array [
  <h1
    className="text-center my-5"
  >
    Update your account
  </h1>,
  <form
    onSubmit={[Function]}
  >
    <label>
      Username: 
    </label>
    <input
      className="form-control my-2"
      name="username"
      onChange={[Function]}
      type="text"
    />
    <label>
      Password: 
    </label>
     
    <input
      className="form-control my-2"
      name="password"
      onChange={[Function]}
      type="text"
    />
    <label>
      Email: 
    </label>
    <input
      className="form-control my-3"
      name="reg_email"
      onChange={[Function]}
      type="text"
    />
    <label>
      Phone: 
    </label>
    <input
      className="form-control my-3"
      name="reg_phone"
      onChange={[Function]}
      type="text"
    />
    <label>
      Cattery name: 
    </label>
    <input
      className="form-control my-3"
      name="cattery_name"
      onChange={[Function]}
      type="text"
    />
    <label>
      Organization: 
    </label>
    <input
      className="form-control my-3"
      name="organization"
      onChange={[Function]}
      type="text"
    />
    <label>
      Owner name: 
    </label>
    <input
      className="form-control my-3"
      name="owner_name"
      onChange={[Function]}
      type="text"
    />
    <label>
      City: 
    </label>
    <input
      className="form-control my-3"
      name="city"
      onChange={[Function]}
      type="text"
    />
    <button
      className="btn btn-success btn-block"
    >
      Update
    </button>
  </form>,
]
`;

exports[`matches snapshot of update parent info 1`] = `
Array [
  <h1
    className="text-center my-5"
  >
    Update your account
  </h1>,
  <form
    onSubmit={[Function]}
  >
    <label>
      Username: 
    </label>
    <input
      className="form-control my-2"
      name="username"
      onChange={[Function]}
      type="text"
    />
    <label>
      Password: 
    </label>
     
    <input
      className="form-control my-2"
      name="password"
      onChange={[Function]}
      type="text"
    />
    <label>
      Email: 
    </label>
    <input
      className="form-control my-3"
      name="reg_email"
      onChange={[Function]}
      type="text"
    />
    <label>
      Phone: 
    </label>
    <input
      className="form-control my-3"
      name="reg_phone"
      onChange={[Function]}
      type="text"
    />
    <label>
      Preferred name: 
    </label>
    <input
      className="form-control my-3"
      name="preferred_name"
      onChange={[Function]}
      type="text"
    />
    <button
      className="btn btn-success btn-block"
    >
      Update
    </button>
  </form>,
]
`;
